server/api/utils/clients/async_nuclio.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import copy
+import urllib.parse
+
+import aiohttp
+
+import mlrun.common.schemas
+import mlrun.utils
+from mlrun.utils import logger
+
+NUCLIO_API_SESSIONS_ENDPOINT = "/api/sessions/"
+NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/"
+API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
+NUCLIO_PROJECT_NAME_HEADER = "X-Nuclio-Project-Name"
+
+
+class Client:
+    def __init__(self, auth_info: mlrun.common.schemas.AuthInfo):
+        self._async_session = None
+        self._auth = aiohttp.BasicAuth(auth_info.username, auth_info.session)
+        self._logger = logger.get_child("nuclio-client")
+        self.nuclio_dashboard_url = mlrun.mlconf.nuclio_dashboard_url
+
+    async def _ensure_async_session(self):
+        if not self._async_session:
+            self._async_session = mlrun.utils.AsyncClientWithRetry(
+                retry_on_exception=mlrun.mlconf.httpdb.projects.retry_leader_request_on_exception
+                == mlrun.common.schemas.HTTPSessionRetryMode.enabled.value,
+                logger=logger,
+            )
+
+    async def _send_request_to_api(
+        self, method, url, path="/", error_message: str = "", **kwargs
+    ):
+        await self._ensure_async_session()
+        response = await self._async_session.request(
+            method=method,
+            url=urllib.parse.urljoin(url, path),
+            verify_ssl=False,
```

Comment:
communication should be internal (k8s service) no need for https thus no need verify_ssl